wallet/db.c

```diff
@@ -200,6 +200,10 @@ char *dbmigrations[] = {
     "DELETE FROM peers WHERE id NOT IN (SELECT peer_id FROM channels);",
     /* The ONCHAIND_CHEATED/THEIR_UNILATERAL/OUR_UNILATERAL/MUTUAL are now one */
     "UPDATE channels SET STATE = 8 WHERE state > 8;",
+    /* What do we think the head of the blockchain looks like? Used
+     * primarily to track confirmations across restarts and making
+     * sure we handle reorgs correctly. */
+    "CREATE TABLE blocks (height INT, hash BLOB, prev_hash BLOB, UNIQUE(height));",
```

Comment:
The `UNIQUE(height)` constraint implies that a reorg deletes entries in this table, right?

Reply:
It does, yes, and it's also what I'm using to set `spend_height` and `confirmation_height` to `NULL` in case of a reorg.

Reply:
The idea is to only track what is actually the valid chain, and have a rollback to the fork point in case of a reorg.